pkg/action/scan.go

```diff
@@ -70,11 +71,25 @@ func scanSinglePath(ctx context.Context, c malcontent.Config, path string, ruleF
 	}
 	logger = logger.With("mime", mime)
 
-	fc, err := os.ReadFile(path)
+	f, err := os.Open(path)
 	if err != nil {
 		return nil, err
 	}
 
+	fi, err := f.Stat()
+	if err != nil {
+		return nil, err
+	}
+
+	size := fi.Size()
+	fc := make([]byte, size)
+
+	_, err = io.ReadFull(f, fc)
+	if err != nil {
+		return nil, err
+	}
+	f.Close()
```

Comment:
```suggestion
	f, err := os.Open(path)
	if err != nil {
		return nil, err
	}
	defer f.Close()

	fi, err := f.Stat()
	if err != nil {
		return nil, err
	}

	size := fi.Size()
	fc := make([]byte, size)

	if _, err := io.ReadFull(f, fc); err != nil {
		return nil, err
	}
```

Reply:
Move the close into a defer further up - this way the file will be closed even if you return early from the other error checks.